Client/core/ServerBrowser/CServerBrowser.cpp

```diff
@@ -1078,21 +1078,21 @@ void CServerBrowser::AddServerToList(CServerListItem* pServer, const ServerBrows
             pServer->iRowIndex = iIndex;
         }
 
-        const SString strVersion = !bIncludeOtherVersions ? "" : pServer->strVersion;
+        const char*   szVersion = !bIncludeOtherVersions ? "" : pServer->strVersion.c_str();
         const SString strVersionSortKey = pServer->strVersionSortKey + pServer->strTieBreakSortKey;
 
-        const SString strPlayers = pServer->nMaxPlayers == 0 ? "" : SString("%d / %d", pServer->nPlayers, pServer->nMaxPlayers);
+        const char*   szPlayers = pServer->nMaxPlayers == 0 ? "" : SString("%d / %d", pServer->nPlayers, pServer->nMaxPlayers).c_str();
```

Comment:
You create a `SString` on the right side, and take the pointer to the internal heap allocated string (though small string optimization might work here and you would get a stack pointer instead), but after this statement/line, the SString will go through the destructor and your `szPlayers` pointer will point to freed memory.